### ISSUE TITLE
[Feature Request]: Defining values for Grouping Variable

### ISSUE THREAD

**Issue by PsyTechMMU** [jasp-stats/jasp-issues#643]
* Enhancement: Ability to define which values are to be used from the Grouping Variable
* Purpose: Conducting independent-samples pairwise-based analyses on datasets with >2 groups or labelled levels
* Use-case: e.g., independent-samples t-test comparing groups 1 and 3 (when grouping variable includes 3 levels)

**Is your feature request related to a problem? Please describe.**
Independent-samples pairwise tests (e.g., independent t-test) cannot be computed in datasets with >2 levels in the grouping variable (without awkward workarounds).

**Describe the solution you'd like**
Ability to define which two levels of the grouping variable are to be used in the analysis.

**Describe alternatives you've considered**
Filtering, but it affects all analyses, and creating multiple dummy variables, but this is tedious and (should be) unnecessary.

**Comment by FransMeerhoff** [comment]
Hi @vandenman ,

Could you have a look at this one, or re-assign it to another?
Cheers Frans

**Comment by AlexanderLyNL** [comment]
This should be fixed automatically with the general plans regarding data editing/filtering. 

I have thought about this, but this could lead to idiosyncratic confusions, and it'll be hard to track computationally, if we fix specific filterings at specific levels of an analysis.

**Comment by PsyTechMMU** [comment]
Without this, is there simply no way (in a single JASP file) to conduct pairwise tests using a grouping variable with >2 values?

**Comment by AlexanderLyNL** [comment]
You can do an ANOVA, and if you planned certain comparisons, you can add contrasts. If you did not plan them, you can do post-hoc tests, and the advantage is that you can correct for multiple comparisons. However, I don't think that post-hoc tests have much inferential meaning other than being exploratory.

**Comment by PsyTechMMU** [comment]
> You can do an ANOVA, and if you planned certain comparisons, you can add contrasts. If you did not plan them, you can do post-hoc tests, and the advantage is that you can correct for multiple comparisons. However, I don't think that post-hoc tests have much inferential meaning other than being exploratory.

Thanks for replying and suggesting that option, but it's really not the same thing. One of the main use-cases for this feature wouldn't even involve ANOVA e.g., just a simple independent-samples t-test between two groups (when the grouping variable includes >2 different values).

(By the way, there's no option for contrasts to be corrected for multiple comparisons, is there?)

**Comment by tomtomme** [comment]
@PsyTechMMU 
If you do a custom contrast along the ANOVA, you can select the two levels you want to compare. And there would be no need for a p-value correction then.

However, for teaching purposes, I see this feature request as beneficial. Because the math of a t-Test is so much simpler, it is still standard to teach it. Otherwise we would just start with teaching ANOVA or GLM.
And SPSS has proven, that it can be done. There you can select which 2 levels to compare, without filtering etc.

**Comment by tomtomme** [comment]
@iandol 
true. But this one is probably much easier / faster to implement than #1330 which would solve this in a much broader sense. So I would just wait on which is closed faster and then close the other one too. So thanks for the hint. I will mark this as duplicate now but leave it open.

**Comment by tomtomme** [comment]
@PsyTechMMU @iandol 
There is an upcoming feature that kind of solves this!
See: https://github.com/jasp-stats/jasp-desktop/pull/5789

This pull will make it feasible to remove levels from data per variable very easily without filtering over the complete data set. So if you have a categorial with 3 levels you would probably duplicate it and then remove the third level to use this in the t-Test. 

You can check this feature out via unstable nightlies here: https://github.com/jasp-stats/jasp-desktop/pull/5789

So I am closing this for now. Please reopen if you do think this does not solve this problem or of course also report back here if this works for you!
